crates/polars-plan/src/plans/optimizer/projection_pushdown/projection.rs

```diff
@@ -59,110 +20,144 @@ pub(super) fn process_projection(
 ) -> PolarsResult<IR> {
     let mut local_projection = Vec::with_capacity(exprs.len());
 
-    // path for `SELECT count(*) FROM`
+    // Special path for `SELECT count(*) FROM`
     // as there would be no projections and we would read
     // the whole file while we only want the count
     if exprs.len() == 1 && is_count(exprs[0].node(), expr_arena) {
         let input_schema = lp_arena.get(input).schema(lp_arena);
         let expr = if input_schema.is_empty() {
             // If the input schema is empty, we should just project
             // ourselves
-            Some(exprs[0].node())
+            exprs[0].node()
         } else {
             // Select the last column projection.
-            let mut name = None;
-            for (_, plan) in (&*lp_arena).iter(input) {
```

Comment:
I removed the LP traversal because I don't think it's needed - if the upper node isn't one of the `Select / HStack / DataframeScan` nodes then it should be safe to take any column